codes/codes.go

```diff
@@ -226,7 +227,11 @@ func (c *Code) UnmarshalJSON(b []byte) error {
 	if c == nil {
 		return fmt.Errorf("nil receiver passed to UnmarshalJSON")
 	}
+	return c.UnmarshalText(bytes.Trim(b, `"`))
```

Comment:
This seems generally fine, but I have a couple changes to request:

1. JSON parsing was loosened by this change.  `"NOT_FOUND` and `"""NOT_FOUND"""` would both parse with this.  Let's make sure this remains as strict as before.

2. `UnmarshalText` probably shouldn't be able to handle integers; let's start with not supporting them and add them later only if we decide it's necessary.

3. It seems unusual to have an `UnmarshalText` without a `MarshalText` or a definition of the format somewhere.  Notably, this is not the inverse of `String` (which I think was a mistake to define the way it is).  Maybe add a `MarshalText` for symmetry?  Or document the format for `b` in the comment on `UnmarshalText` -- I don't feel too strongly either way.

One way to implement (1) and (2) above:

- In `UnmarshalJSON`: attempt to use `json.Unmarshal` into a `string` (or a `*string`, perhaps, to remove the `null` check).
  - If it succeeds, call `UnmarshalText` with the result.  If using `*string` and `nil`, return.
  - If it fails, fall back to the check for an integer

Reply:
@dfawley Good points. I'll make the changes, and I'll add MarshalText for symmetry - I like that!

Reply:
@dfawley I have some time set aside to hack on this in the next few days - just keeping you up to date.